substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -150,6 +150,9 @@ impl<T: Config> Pallet<T> {
 			// Already checked that this won't overflow by entry condition.
 			let value = old_total.defensive_saturating_sub(new_total);
 			Self::deposit_event(Event::<T>::Withdrawn { stash, amount: value });
+
+			// notify listeners.
+			T::EventListeners::on_withdraw(controller, old_total.saturating_sub(new_total));
```

Comment:
Maybe just clone and reuse `value`. 
```suggestion
			T::EventListeners::on_withdraw(controller, value);
```